public_html/wp-content/themes/pattern-directory/src/components/report-pattern-button/index.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Icon, check } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import ReportPatternModal from '../report-pattern-modal';
+
+const ReportPatternButton = ( { postId, loggedIn, userHasReported } ) => {
+	const [ showModal, setShowModal ] = useState( false );
+	const [ hasSubmitted, setHasSubmitted ] = useState( false );
+	const alreadySubmitted = userHasReported || hasSubmitted;
+
+	if ( alreadySubmitted ) {
+		return (
+			<p className="pattern-report-button__copy">
+				<Icon icon={ check } />
+				{ __( "You've reported this pattern", 'wporg-patterns' ) }
+			</p>
+		);
+	}
+
+	if ( ! loggedIn ) {
+		return (
+			<p className="pattern-report-button__copy">
+				<a href="/wp-login.php">{ __( 'Login to report this pattern', 'wporg-patterns' ) }</a>
```

Comment:
The URL here should have a `redirect_to` query parameter with the current URL as the value so that after the user logs in they come back to the pattern page (otherwise they end up in WP Admin).

In PHP there's `wp_login_url()` which sets this up for you, but I'm not seeing an equivalent on the Gutenberg side...

Reply:
Ah great! I didn't know there was an underlying redirect system.

We can use `addQueryArgs` to add the redirect.